File: src/main/java/ca/yyx/hu/decoder/AudioDecoder.java
package ca.yyx.hu.decoder;

import android.util.SparseArray;

/**
 * @author algavris
 * @date 28/04/2016.
 */
public class AudioDecoder {
    public static final int SAMPLE_RATE_HZ_48 = 48000;
    public static final int SAMPLE_RATE_HZ_16 = 16000;

    private SparseArray<AudioTrackWrapper> mAudioTracks = new SparseArray<>(3);

    public AudioTrackWrapper getTrack(int channel)
    {
        return mAudioTracks.get(channel);
    }

    public void decode(int channel, byte[] buffer, int offset, int size) {
        AudioTrackWrapper audioTrack = mAudioTracks.get(channel);
        audioTrack.write(buffer, offset, size);
    }

    public void stop() {
        for (int i = 0; i < mAudioTracks.size(); i++)
        {
            stop(mAudioTracks.keyAt(i));
        }
    }

    public void stop(int chan) {
        AudioTrackWrapper audioTrack = mAudioTracks.get(chan);
        if (audioTrack != null) {
            audioTrack.stop();
            mAudioTracks.put(chan, null);
        }
    }

    public void start(int channel, int stream, int sampleRate, int numberOfBits, int numberOfChannels) {
        AudioTrackWrapper thread = new AudioTrackWrapper(stream, sampleRate, numberOfBits, numberOfChannels);
        mAudioTracks.put(channel, thread);
    }
}

File: src/main/java/ca/yyx/hu/decoder/AudioDecoder.kt
package ca.yyx.hu.decoder

import android.util.SparseArray

/**
 * @author algavris
 * *
 * @date 28/04/2016.
 */
class AudioDecoder {

    private val mAudioTracks = SparseArray<AudioTrackWrapper>(3)

    fun getTrack(channel: Int): AudioTrackWrapper? {
        return mAudioTracks.get(channel)
    }

    fun decode(channel: Int, buffer: ByteArray, offset: Int, size: Int) {
        val audioTrack = mAudioTracks.get(channel)
        audioTrack.write(buffer, offset, size)
    }

    fun stop() {
        for (i in 0..mAudioTracks.size() - 1) {
            stop(mAudioTracks.keyAt(i))
        }
    }

    fun stop(chan: Int) {
        val audioTrack = mAudioTracks.get(chan)
        audioTrack?.stop()
        mAudioTracks.put(chan, null)
    }

    fun start(channel: Int, stream: Int, sampleRate: Int, numberOfBits: Int, numberOfChannels: Int) {
        val thread = AudioTrackWrapper(stream, sampleRate, numberOfBits, numberOfChannels)
        mAudioTracks.put(channel, thread)
    }

    companion object {
        const val SAMPLE_RATE_HZ_48 = 48000
        const val SAMPLE_RATE_HZ_16 = 16000
    }
}
